src/main/kotlin/com/valbaca/gotlin/ch8/Clock2.kt

```diff
@@ -5,15 +5,16 @@ import java.net.ServerSocket
 import java.net.Socket
 import java.time.LocalTime
 import java.time.format.DateTimeFormatter
-import kotlin.time.DurationUnit.SECONDS
-import kotlin.time.toDuration
+import kotlin.time.Duration.Companion.seconds
 
 /*
 p222 NonBlocking TCP Clock Server using coroutines
 https://github.com/adonovan/gopl.io/tree/master/ch8/clock2
 */
 private val clockFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern("HH:mm:ss")
-fun main() = runBlocking(Dispatchers.IO) {
+private val printUncaughtException = CoroutineExceptionHandler { _, ex -> ex.printStackTrace() }
+
+suspend fun main() = supervisorScope {
```

Comment:
Normally, when a child job fails, the entire scope fails. Supervisor scope will prevent a failing child job from killing the scope.